extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/endpoint/BasicAuthenticatorResource.java

```diff
@@ -102,6 +106,7 @@ public Response getAllUsers(
       @PathParam("authenticatorName") final String authenticatorName
   )
   {
+    authValidator.validateAuthenticatorName(authenticatorName);
```

Comment:
Can this check be added in `BasicSecurityResourceFilter` instead of adding it everywhere? `BasicSecurityResourceFilter` is used by both `BasicAuthenticationResource` and `BasicAuthorizerResource`.

Reply:
I thought about adding this to a filter like I did here - https://github.com/apache/druid/pull/10106/files/66b4f28bfb52acaf3243a2277167c5ae1cec322b#diff-d816117984574d3000bb1c8b6faaccb0

However, this resource has many different paths that pre-fix the authenticator/ authorizer name -`/db/{authenticatorName}`, `/listen/{authenticatorName}`, `/listen/users/{authorizerName}` and `/listen/groupMappings/{authorizerName}`.

So I figured this would be error-prone, since there isn't a consistent way to access the authenticator / authorizer name.

I think doing this in a filter makes the code cleaner, but I worry about missing an endpoint or getting the logic correct when picking the authorizer name from the path.

Reply:
I see. I was worrying about that this way is easy to forget when we add a new API, but it seems like the other way is error-prone as well.

Reply:
yeah :(